movement-migration/framework_upgrade.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+# Script usage
+usage() {
+    echo "Usage: $0 -k <private_key_file> -u <url> -e <initial_epoch> -t <initial_timestamp>"
+    echo ""
+    echo "Required arguments:"
+    echo "  -k    Path to the private key file"
+    echo "  -u    URL of the network (e.g., http://127.0.0.1:8080)"
+    echo "  -e    Initial epoch (e.g., 2111770). Get this from the node /v1 url. If you already set the value, DO NOT set it again."
+    echo "  -t    Initial timestamp (e.g., 1762293093318227). Get this from the node /v1 url. If you already set the value, DO NOT set it again"
+    echo ""
+    echo "Example:"
+    echo "  $0 -k ../private_key_testnet.key -u http://127.0.0.1:8080 -e 2111770 -t 1762293093318227"
+    exit 1
+}
+
+# Parse command-line arguments
+PRIVATE_KEY_FILE=""
+URL=""
+INITIAL_EPOCH=""
+INITIAL_TIMESTAMP=""
+
+while getopts "k:u:e:t:h" opt; do
+    case ${opt} in
+        k )
+            PRIVATE_KEY_FILE="$OPTARG"
+            ;;
+        u )
+            URL="$OPTARG"
+            ;;
+        e )
+            INITIAL_EPOCH="$OPTARG"
+            ;;
+        t )
+            INITIAL_TIMESTAMP="$OPTARG"
+            ;;
+        h )
+            usage
+            ;;
+        \? )
+            echo "Invalid option: -$OPTARG" >&2
+            usage
+            ;;
+    esac
+done
+
+# Check if required arguments are provided
+if [ -z "$PRIVATE_KEY_FILE" ] || [ -z "$URL" ] || [ -z "$INITIAL_EPOCH" ] || [ -z "$INITIAL_TIMESTAMP" ]; then
+    echo "Error: All arguments (private key file, URL, initial epoch, and initial timestamp) are required."
+    usage
+fi
+
+# Validate that the initial epoch and timestamp are numeric
+if ! [[ "$INITIAL_EPOCH" =~ ^[0-9]+$ ]]; then
+    echo "Error: Initial epoch must be a numeric value"
+    exit 1
+fi
+
+if ! [[ "$INITIAL_TIMESTAMP" =~ ^[0-9]+$ ]]; then
+    echo "Error: Initial timestamp must be a numeric value"
+    exit 1
+fi
+
+# Verify that the private key file exists
+if [ ! -f "$PRIVATE_KEY_FILE" ]; then
+    echo "Error: Private key file not found: $PRIVATE_KEY_FILE"
+    exit 1
+fi
+
+# Enable strict error handling - exit on any command failure
+set -e
+
+echo "Starting framework upgrade..."
+echo "Private key file: $PRIVATE_KEY_FILE"
+echo "URL: $URL"
+echo "Initial epoch: $INITIAL_EPOCH"
+echo "Initial timestamp: $INITIAL_TIMESTAMP"
+echo ""
+
+# update the framework for epoch duration update
+cargo run -p movement move run-script --script-path  ./movement-migration/update_reconfiguration/clone_proposals/sources/vX.YY.Z/proposal_1_upgrade_framework/0-move-stdlib.move  --bytecode-version 6 --sender-account 0xa550c18 --private-key-file "$PRIVATE_KEY_FILE" --url "$URL" --assume-yes
```

Comment:
Cargo run initiate a build of the movement cli. It takes time. For the migration, we have to:
 - prebuild the aptos cli before the migration on all instance where this script is executed
 - or install the cli on each instance.